libs/ngeth/src/lib/wallet/localstorage-vault.ts

```diff
@@ -0,0 +1,59 @@
+import { Vault, KeyVault } from './vault';
+
+export class LocalStorageVault extends Vault {
+
+  private storage: Storage;
+
+  constructor() {
+    super();
+    try {
+      this.storage = (window as any).localStorage;
+    } catch(err) {
+      throw new Error('This environement doesn\'t support a Local Storage');
+    }
+  }
+
+  get(accounts: 'accounts'): Promise<string[]>;
+  get(address: 'default'): Promise<string>;
+  // tslint:disable-next-line: unified-signatures
+  get(address: string): Promise<string>;
+  get(key: KeyVault): Promise<string | string[]> {
+    switch(key) {
+      case 'accounts': return JSON.parse(this.storage.getItem('accounts'));
+      case 'default': return JSON.parse(this.storage.getItem('default'));
+      default: return JSON.parse(this.storage.getItem(key));
+    }
+  }
+
+  set(key: 'accounts', accounts: string[]): Promise<any>;
+  set(key: 'default', address: string): Promise<any>;
+  // tslint:disable-next-line: unified-signatures
+  set(address: string, json: string): Promise<any>;
+  set(key: string, payload: string | string[]): Promise<any> {
+    return new Promise((res, rej) => {
```

Comment:
Just add "async" in front of the function, it'll transform the result into a promise (no need to do `new Promise`).